github-actions-scripts/folioci_npmver.sh

```diff
@@ -1,39 +1,100 @@
-#!/bin/bash
+#!/usr/bin/env node
 
-# for testing only
-#BUILD_NUMBER=123
+/**
+ * SUMMARY
+ *
+ * derive a semver-compatible value from ./package.json's version and the build-number
+ *
+ *
+ * USAGE
+ *
+ * JOB_ID=123 ./$0              // output like 11.1.109000000123
+ * JOB_ID=1234 ./$0             // output like 11.1.109000001234
+ * JOB_ID=123 new_ci=true ./$0  // output like 11.1.10990000000123
+ * JOB_ID=1234 new_ci=true ./$0 // output like 11.1.10990000001234
+ *
+ *
+ * DETAILS
+ *
+ * Given package.json contains a version value like `1.2.0` or `2.3.4` and
+ * the environment variable JOB_ID contains a build-number like 456, concoct
+ * a version number like `1.2.109000000456` or `2.3.409000000456` and print it.
+ *
+ * When the environment variable new_ci is truthy, the generated patch value
+ * is padded to 14-16 places with a prefix of ${patch}099. Otherwise, the
+ * generated patch value is padded to 12-14 places with a prefix of ${patch}09.
+ *
+ * Exits 0 on success, 2 if package.json is not found, and 3 in case of any
+ * other error.
+ *
+ * The max patch value from package.json is 900.
+ *
+ * The max JOB_ID is 9_999_999_999 when new_ci is true, 999_999_999 without it.
+ *
+ * Why oh why do we do it this way? A long time ago, on a Jenkins server far
```

Comment:
I like this section. (Although I still don't understand why we ever used `9` instead of `1`, or why the new version used `99` instead of `91`. Oh well.)